garden_ai/app/notebook.py

```diff
@@ -34,6 +37,35 @@
 )
 
 
+class DockerClientSession:
+    def handle_docker_start_failure(self, e: DockerStartFailure):
+        print_err("Garden can't access Docker on your computer.")
+        if e.helpful_explanation:
+            print_err(e.helpful_explanation)
+            print_err(
+                "If that doesn't work, use `garden-ai docker check` to troubleshoot."
+            )
+        else:
+            print_err(
+                "This doesn't look like one of the typical error cases. Printing error from Docker:"
+            )
+            typer.echo(e.original_exception)
+        raise typer.Exit(1)
+
+    # We're most likely to see this error raised from get_docker_client,
+    def __enter__(self):
+        try:
+            return get_docker_client()
+        except DockerStartFailure as e:
+            self.handle_docker_start_failure(e)
+
+    # but if the user's Docker daemon shuts down partway through the session
+    # we'll catch that here.
+    def __exit__(self, exc_type, exc_val, exc_tb):
```

Comment:
fwiw I tried quitting the docker app while a notebook was running and this didn't trip, but I think the approach is sound. probably just a matter of hunting down which other various `exc_type`s we can get into here but that's outside the scope for this PR imo

Reply:
Yeah. As of this PR the decorator just catches startup errors. So if Docker turns off in the middle of a notebook session it wouldn't catch that. That said, totally, the hope is that as we see other common and troublesome cases we can add them in here